ocaml/xapi-idl/xen/device_number.ml

```diff
@@ -76,13 +78,12 @@ let of_xenstore_int x =
         (Floppy, (x lsr 4) land ((1 lsl 4) - 1), x land ((1 lsl 4) - 1))
     | n ->
         let idx =
-          snd
-            (List.fold_left
```

Comment:
We could build a Map in a global var that gives you the answer directly. The table is not going to change.

Reply:
I've replaced the list with an array and an intmap for faster accesses